packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -331,6 +340,10 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
     pageTo(_firstRowIndex + widget.rowsPerPage);
   }
 
+  void _handleLast() {
+    pageTo(((_rowCount - 1) / widget.rowsPerPage).floor() * widget.rowsPerPage);
```

Comment:
I'm having trouble wrapping my head around this. Can you explain?

Reply:
I actually got this algo from here: https://github.com/DataTables/DataTables/blob/de73685996c6c666e1850668c72e34d492f22150/media/js/jquery.dataTables.js#L4987 which is a well established library for DataTables which has been around for over 10 years.